packages/openapi3/test/http-specs/utils/spec-snapshot-testing.ts

```diff
@@ -0,0 +1,195 @@
+        const outputPath = resolvePath(config.outputDir, snapshotPath);
+        let existingContent;
+        try {
+          existingContent = await readFile(outputPath);
+        } catch (e: unknown) {
+          if (isEnoentError(e)) {
+            fail(`Snapshot "${outputPath}" is missing. Run with RECORD=true to regenerate it.`);
+          }
+          throw e;
+        }
+        context.registerSnapshot(resolvePath(spec.name, snapshotPath));
+        deepStrictEqual(JSON.parse(existingContent.toString()), content);
+        //strictEqual(prettierOutput(JSON.stringify(content, null, 2)), existingContent.toString());
+      }
+    }
+  });
+}
+
+interface SpecSnapshotTestHost extends CompilerHost {
+  outputs: Map<string, string>;
+}
+
+function createSpecTestHost(): SpecSnapshotTestHost {
+  const outputs = new Map<string, string>();
+  return {
+    ...NodeHost,
+    outputs,
+    mkdirp: (path: string) => Promise.resolve(path),
+    writeFile: async (path: string, content: string) => {
+      outputs.set(path, content);
+    },
+  };
+}
+
+interface Spec {
+  name: string;
+  /** Spec folder */
+  fullPath: string;
+}
+
+function resolveSpecs(config: SpecSnapshotTestOptions): Spec[] {
+  const specs: Spec[] = [];
+  const excludes = new Set(config.exclude);
+  walk("");
+  return specs;
+
+  function walk(relativeDir: string) {
+    if (excludes.has(relativeDir)) {
+      return;
+    }
+    const fullDir = joinPaths(config.specDir, relativeDir);
+    for (const entry of readdirSync(fullDir, { withFileTypes: true })) {
+      if (entry.isDirectory()) {
+        walk(joinPaths(relativeDir, entry.name));
+      } else if (relativeDir && entry.name === "main.tsp") {
+        specs.push({
+          name: relativeDir,
+          fullPath: joinPaths(config.specDir, relativeDir),
+        });
+      }
+    }
+  }
+}
+
+function isEnoentError(e: unknown): e is { code: "ENOENT" } {
+  return typeof e === "object" && e !== null && "code" in e;
+}
+
+export async function openApiForFile(spec: Spec, options: OpenAPI3EmitterOptions = {}) {
+  const host = createSpecTestHost();
+  const program = await compile(host, spec.fullPath, {
+    noEmit: false,
+    emit: ["@typespec/openapi3"],
+    options: { "@typespec/openapi3": { ...options, "file-type": "json" } },
+  });
+  expectDiagnosticEmpty(program.diagnostics);
+  const openApiVersion = options["openapi-versions"]?.[0] ?? "3.0.0";
+  const output: any = {};
+  for (const [path, content] of host.outputs.entries()) {
+    const snapshotPath = resolvePath(openApiVersion, spec.name, path);
+    const jsonContent = JSON.parse(content);
+    //await validateOpenAPI3(jsonContent);
+
+    output[snapshotPath] = jsonContent;
+  }
+  return output;
+}
+
+export async function markCoverage(path: string, options: Record<string, any>) {
+  const BASE_PATH = "http://localhost:3000";
+
+  try {
+    await fetch(BASE_PATH + path, options);
```

Comment:
you should use the response to verify the generated OpenAPI3 document

Reply:
> you should use the response to verify whether the response matches the generated OpenAPI3 document
@timotheeguerin , I believe it's enough to 
- use the response to verify whether the response matches the generated OpenAPI3 document
- verify the entire generated OpenAPI3 document
for cadl-ranch (now named spectors???) cases. Correct?